### PR TITLE
Add chain upgrade jobs for arm and multiarch

### PR DESCRIPTION
**Pull request by shellyyang1989**
cc @jianlinliu @jiajliu @JianLi-RH for review. Thank you.

### CODE REVIEW

**Comment by shellyyang1989** [comment]
cc @aleskandro for review. Thank you.

**Comment by shellyyang1989** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-multi-nightly-4.15-upgrade-from-stable-4.13-azure-ipi-ovn-ipsec-arm-mixarch-f14

**Comment by shellyyang1989** [comment]
Refers [OCPQE-17848](https://issues.redhat.com/browse/OCPQE-17848)

**Comment by aleskandro** [comment]
Hi @shellyyang1989 what if we run the AWS and Azure ones by using the multi payload and `ADDITIONAL_WORKERS: 0`. We'd like to reduce the amount of single-arch jobs time by time: see https://issues.redhat.com/browse/OCPQE-17309

**Comment by shellyyang1989** [comment]
> Hi @shellyyang1989 what if we run the AWS and Azure ones by using the multi payload and `ADDITIONAL_WORKERS: 0`. We'd like to reduce the amount of single-arch jobs time by time: see https://issues.redhat.com/browse/OCPQE-17309

Hmm, I'm not sure I get you. Per OCP release notes, multi-arch is TP on 4.12 where upgrade is not supported. So I'm adding 4.13 to 4.15 jobs on azure and aws for multi-arch. If your concern is about the `ADDITIONAL_WORKERS: 0`, It's fine with me to change it to `0`

**Comment by shellyyang1989** [comment]
The 4.10 to 4.15 [rehearse](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/46114/rehearse-46114-periodic-ci-openshift-openshift-tests-private-release-4.15-arm64-nightly-4.15-upgrade-from-stable-4.10-aws-ipi-ovn-ipsec-f14/1729435673334648832) failed due to [OCPBUGS-22293](https://issues.redhat.com/browse/OCPBUGS-22293). It's expected.

```
   Message:               Error while updating operator configuration: could not apply (apps/v1, Kind=DaemonSet) openshift-ovn-kubernetes/ovnkube-master: failed to apply / update (apps/v1, Kind=DaemonSet) openshift-ovn-kubernetes/ovnkube-master: DaemonSet.apps "ovnkube-master" is invalid: [spec.template.spec.containers[1].lifecycle.preStop: Required value: must specify a handler type, spec.template.spec.containers[3].lifecycle.preStop: Required value: must specify a handler type]
```


The 4.13 to 4.15 [rehearse](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/46114/rehearse-46114-periodic-ci-openshift-openshift-tests-private-release-4.15-multi-nightly-4.15-upgrade-from-stable-4.13-azure-ipi-ovn-ipsec-arm-mixarch-f14/1729435471823507456) passed on upgrade
```
INFO[2023-11-28T14:08:36Z] Running step azure-ipi-ovn-ipsec-arm-mixarch-f14-cucushift-upgrade-toimage. 
INFO[2023-11-28T15:19:36Z] Step azure-ipi-ovn-ipsec-arm-mixarch-f14-cucushift-upgrade-toimage succeeded after 1h11m0s. 
```

**Comment by aleskandro** [comment]
> > Hi @shellyyang1989 what if we run the AWS and Azure ones by using the multi payload and `ADDITIONAL_WORKERS: 0`. We'd like to reduce the amount of single-arch jobs time by time: see https://issues.redhat.com/browse/OCPQE-17309
> 
> Hmm, I'm not sure I get you. Per OCP release notes, multi-arch is TP on 4.12 where upgrade is not supported. So I'm adding 4.13 to 4.15 jobs on azure and aws for multi-arch. If your concern is about the `ADDITIONAL_WORKERS: 0`, It's fine with me to change it to `0`

You're right. That's all fine to me as is.

Thanks @shellyyang1989 

/lgtm

**Comment by shellyyang1989** [comment]
Thank you @aleskandro 

@liangxia Can you please help review and merge?

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__arm64-nightly-4.15-upgrade-from-stable-4.10.yaml [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+      OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE: release:arm64-target
+    env:
+      BASE_DOMAIN: qe.devcluster.openshift.com
+      COMPUTE_NODE_TYPE: m6g.xlarge
+      CONTROL_PLANE_INSTANCE_TYPE: m6g.xlarge
+      OCP_ARCH: arm64
+      UPGRADE_CHECK_RUN_TAGS: '@arm64 and @upgrade-check and @4.14 and @aws-ipi and
+        not @fips'
+      UPGRADE_PRE_RUN_TAGS: '@arm64 and @upgrade-prepare and @4.14 and @aws-ipi and
+        not @fips'
```
[the remark this answers is not in the document]
Comment (shellyyang1989) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__arm64-nightly-4.15-upgrade-from-stable-4.10.yaml:107]: For 4.12+, ovn is the default network provider, while sdn is the default provider for 4.12-

[the remark this answers is not in the document]
Comment (liangxia) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__arm64-nightly-4.15-upgrade-from-stable-4.10.yaml:107]: For jobs added in this PR, I can see all of them are OVN jobs.

[the remark this answers is not in the document]
Comment (shellyyang1989) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__arm64-nightly-4.15-upgrade-from-stable-4.10.yaml:107]: Good catch! Thank you.

**Pull request merged**